src/app/clusters/ota-requestor/GenericOTARequestorDriver.h

```diff
@@ -87,7 +87,7 @@ class GenericOTARequestorDriver : public OTARequestorDriver
     uint32_t mPeriodicQueryTimeInterval = (24 * 60 * 60); // Timeout for querying providers on the default OTA provider list
 
     using ProviderLocationType = app::Clusters::OtaSoftwareUpdateRequestor::Structs::ProviderLocation::Type;
-    ProviderLocationType mLastUsedProvider; // Provider location used for the last query or update
+    Optional<ProviderLocationType> mLastUsedProvider; // Provider location used for the last query or update
```

Comment:
Since this is not stored, it means the first item in the list is always preferred on a cold boot, rather than last successful. Perhaps storing last successful entry to see mLastUsedProvider at init would be worthwhile. This can be done as a follow-up, since it's an optimization

Reply:
I think the idea would be to persist the lastUsedProvider in the flash, since the requestor will have to send the NotifyUpdateApplied message to this provider, post-installation of the new firmware. But yes, that would be a follow-up.